sound/soc/sof/control.c

```diff
@@ -375,7 +375,7 @@ int snd_sof_bytes_put(struct snd_kcontrol *kcontrol,
 	}
 
 	/* copy from kcontrol */
-	memcpy(data, ucontrol->value.bytes.data, data->size);
+	memcpy(data, ucontrol->value.bytes.data, data->size + sizeof(*data));
```

Comment:
this doesn't look fully correct, the test at line 361 should be updated as well to match what is done for get?

Reply:
@keyonjie this change seems OK. 
But This API only supports up to 512 bytes max data size. We should be using the bytes_ext IO for all binary data. So I'd like to know what issue this change fixes.

Reply:
> this doesn't look fully correct, the test at line 361 should be updated as well to match what is done for get?

Make sense, let me correct it.

Reply:
> @keyonjie this change seems OK.
> But This API only supports up to 512 bytes max data size. We should be using the bytes_ext IO for all binary data. So I'd like to know what issue this change fixes.

We are now using this API for test_detect component blob(smaller than 512, about 64 Bytes), we hit wrong bytes kcontrol put issue there.

You are correct, we will use bytes_ext for > 512B cases in other detector if needed.